### ISSUE TITLE
Allow to stop ongoing backup

### ISSUE THREAD

**Issue by DJaeger** [jensstein/oandbackup#102]
From #97:

DJaeger commented 2 days ago

How about creating the new app backups in a folder with attached _temp to the name and if ready attach a _old to the old one, rename the new one to only the appname, without attachment, and than remove the _old folder.
If it is canceled, simply remove the _temp folder.
With that the replacement of the already available data with the new data, would take nearly no time and will always keep the old data until it is really ready.

jensstein commented 2 days ago

yes, that is my immediate thought. but then apps with large amounts of data quickly become a problem. what if some app has a data size of 1GB or something and the storage area simply doesn't have enough space for both the temporary files and the original backups.
that may just be the way it is, so maybe there could be a check for available storage and then the cancel functionality could just be disabled on apps with too much data. but if you have another suggestion i'd very much like to hear it.

DJaeger commented a day ago

The app should anyway check the available storage before starting to backup an app.

There could also be a setting for "safe copying", which enables this way.
So the user can choose between the safe way, which consumes more space or the lesser safe way, which uses lesser space.
The safe way should be the default, as it can also prevent broken backups by crashing or killed app.

If safe way is activated another option for "reverse currend apps backup on cancel" could be available to allow the user to choose the way to cancel (reverse current app or stop after current app).
If safe way is disabled, never use reversing, but stop after current app.

To disable the cancel button is a bad way, as the user have to wait to stop until the next not to big app's backup is started.
As you sometimes simply wants to stop to prevent the heavy battery consuming, it should be available anytime in anyway.
Think of you have a planned backup every night at 3 AM, but one night you are on a party, it is 3:15 AM, you look on your phone and thinks: "Oh, with this battery usage I will not be able to call a taxi if I want to go home, so I should better stop the backup for tonight".

BTW: I think we should move this "stop button" discussion to a new issue and label it as enhancement.
